### PR TITLE
feat(read-tracking): implement read tracking with basic functionalities

### PR DESCRIPTION
**Pull request by jgmontoya**
## Summary by CodeRabbit

* **New Features**
  * Users can mark messages as read; the system tracks each account’s last-read message per chat.
  * Chat list items now show per-chat unread message counts.

* **Tests**
  * Added integration and unit tests covering marking-as-read, unread-count calculations, and related edge cases.

* **Chores**
  * Database migration added to persist per-chat last-read markers.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: src/whitenoise/chat_list.rs [hunk 1]
```diff
@@ -215,6 +221,19 @@ impl Whitenoise {
...
 
+        // Compute unread counts for all groups
+        let mut unread_counts = HashMap::new();
+        for (group_id, account_group) in &membership_map {
+            let count = AggregatedMessage::count_unread_for_group(
+                group_id,
+                account_group.last_read_message_id.as_ref(),
+                &self.database,
+            )
+            .await
```
Comment (dannym-arx) [inline src/whitenoise/chat_list.rs]: rather than looping, could we make a single query that returns (group_id, count) for every group?

File: src/whitenoise/database/aggregated_messages.rs [hunk 2]
```diff
@@ -479,6 +479,65 @@ impl AggregatedMessage {
...
+                     WHERE mls_group_id = ? AND kind = 9 AND deletion_event_id IS NULL",
+                )
+                .bind(group_id.as_slice())
+                .fetch_one(&database.pool)
+                .await?
+            }
+        };
+
+        Ok(count as usize)
+    }
```
[the remark this answers is not in the document]
Comment (jgmontoya) [reply src/whitenoise/database/aggregated_messages.rs:539]: This is an acceptable edgecase

File: src/whitenoise/chat_list.rs (hunk 1 above)
Comment (jgmontoya) [reply src/whitenoise/chat_list.rs]: I thought it might be too complex but I will give it a try

Comment (mubarakcoded) [reply src/whitenoise/chat_list.rs]: I agree with @dannym-arx, this creates  N+1 query problem. If a user has 50 groups, this means 50 separate database queries, that's going to be really slow. Can we batch this into a single query that returns all unread counts at once?

File: src/whitenoise/chat_list.rs [hunk 3]
```diff
@@ -319,7 +339,16 @@ impl Whitenoise {
...
 
-        // 8. Assemble and return ChatListItem
+        // 8. Compute unread count
+        let unread_count = AggregatedMessage::count_unread_for_group(
+            group_id,
+            account_group.last_read_message_id.as_ref(),
+            &self.database,
+        )
+        .await
+        .unwrap_or(0);
```
Comment (mubarakcoded) [inline src/whitenoise/chat_list.rs]: We will be silently swallowing database errors here if the db query fails for some reason, user might see incorrect unread counts. Should we be propagating this error instead or at minimum logging it?
